### PR TITLE
Password manager auto-fill works in Firefox!

### PR DESCRIPTION
**Pull request by shane-tomlinson**
- Move the email/password form to always be in the DOM. This is a requirement for the Firefox password manager to auto-fill passwords.
- Update all the unit tests.
- Remove the authenticate template, move the contents into dialog.ejs

Open issues:
- There are multiple HTML elements with the "email" and "password" IDs. This will require these elements to be renamed in either the authentication screen or in the other screens. Changes to these IDs will require changes to bidpom.

issue #314

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@6a68, @davehunt, @zacc - Can you weigh in on the process necessary to change element IDs in bidpom?

File: resources/static/dialog/js/modules/authenticate.js [hunk 1]
```diff
@@ -174,12 +182,13 @@ BrowserID.Modules.Authenticate = (function() {
       lastEmail = options.email || "";
 
       var self=this;
-      self.renderDialog("authenticate", {
-        siteName: options.siteName,
-        email: lastEmail
-      });
+
+      dom.addClass(BODY_SELECTOR, AUTHENTICATION_CLASS);
+      dom.setInner(RP_NAME_SELECTOR, options.siteName);
```
Comment (seanmonstar) [inline resources/static/dialog/js/modules/authenticate.js:187]: The `siteName` is escaped before somewhere?

Comment (shane-tomlinson) [inline resources/static/dialog/js/modules/authenticate.js:187]: @seanmonstar - https://github.com/mozilla/browserid/blob/71341882496bf4573335a0f96693b40e15bc2ee9/resources/static/dialog/js/modules/dialog.js#L228

Comment (seanmonstar) [inline resources/static/dialog/js/modules/authenticate.js:187]: I see it, its escaped in `dialog.get()` at the beginning of the request.

**Pull request merged**